src/main/java/org/broadinstitute/hellbender/utils/read/SAMRecordToGATKReadAdapter.java

```diff
@@ -27,6 +27,20 @@ public SAMRecordToGATKReadAdapter( final SAMRecord samRecord ) {
         this.samRecord = samRecord;
     }
 
+    /**
+     * Produces a SAMRecordToGATKReadAdapter wrapping the provided SAMRecord,
+     * and nulls out the header in the encapsulated read. This is useful for
+     * Spark tools, in order to avoid serializing the SAMFileHeader for each
+     * record.
+     *
+     * @param samRecord read to adapt (header will be stripped)
+     * @return SAMRecordToGATKReadAdapter wrapping the headerless samRecord
+     */
+    public static SAMRecordToGATKReadAdapter headerlessReadAdapter( final SAMRecord samRecord ) {
```

Comment:
Why not just do this in the constructor?  It seems confusing to have 2 paths to make the adaptors, one of which uses headers and one which doesn't?